apps/backend/src/reconciliation/reconciliation.service.ts

```diff
@@ -30,4 +59,146 @@ export class ReconciliationService {
       throw err;
     }
   }
+
+  public async getMerchantIds(location_id: number): Promise<number[]> {
+    const merchant_ids = await this.transactionRepo.manager.query(`
+        SELECT DISTINCT "Merchant ID" 
+        FROM public.master_location_data ml 
+        WHERE ml."GARMS Location" = ${location_id} 
+        AND "Type" != 'Bank'
+      `);
+    return merchant_ids.map((itm: any) => parseInt(itm['Merchant ID']));
+  }
+
+  public async getPTIDs(location_id: number): Promise<string[]> {
+    const pt_ids = await this.transactionRepo.manager.query(`
+          SELECT DISTINCT "Location" 
+          FROM public.master_location_data ml 
+          WHERE ml."GARMS Location" = ${location_id}
+        `);
+    return pt_ids.map((itm: any) => itm['Location']);
+  }
+
+  public async getLocationList(): Promise<ILocation[]> {
+    const locations = await this.transactionRepo.manager.query(`
+        SELECT "GARMS Location", description 
+        FROM public.master_location_data ml 
+        WHERE ml."Type" = 'Bank' 
+        ORDER BY "GARMS Location" DESC;
+      `);
+    return locations.map((itm: any) => ({
+      sbc_location: itm['GARMS Location'],
+      office_name: itm.description
+    }));
+  }
+
+  public async getMethod(sbc_code: number) {
+    const payment_method = await this.paymentMethodRepo.findOne({
+      where: { sbc_code }
+    });
+    if (payment_method) {
+      return payment_method.sbc_code;
+    }
+  }
+
+  public async reconcilePOS(date: string, location_id: number) {
```

Comment:
Should we name this `reconcilePOSBySalesLocation`? 

Also, this is a great example of what I call `Layer`